src/Api/Response/ResponseParser.php

```diff
@@ -41,9 +41,9 @@ public function parse($responseData)
 
         try {
             $data = $this->loadXml($responseData);
+            $this->handleLandingPage($data);
```

Comment:
For better testability I would suggest:
* Call `parseLandingPage`.
* Pass the return value (if valid) to `handleLandingPage`.
* Instead of mocking `parseLandingPage` in the test, check if `handleLandingPage` has been called with the correct parameter.